test/darkleaf/di/dependencies_test.clj

```diff
@@ -97,6 +97,21 @@
            (try-ex-data (di/start `parent)))))
 
 
+(defn ^{::di/kind :component} component-no-value
+  [])
+
+
+(defn parent-2
+  [{::syms [component-no-value]}]
+  :done)
+
+
+(t/deftest component-no-value-exception-test
+  (t/is (= {:type  ::di/component-no-value
+            :stack [`component-no-value `parent-2 ::di/implicit-root]}
```

Comment:
You don’t need the stack here. You can identIfy an invalid component just by its key

Reply:
There seem to be two issues. The first is a notification about invalid component. 
The second is about adding extra info such a stack of components.

You can solve the second with the following pattern:

```clojure
(try
  (...)
  (catch RuntimeException ex
    (throw (ex-info "..." {:stack stack} ex))))
```